examples/tfengine/modules/team.hcl

```diff
@@ -47,27 +47,29 @@ template "project_secrets" {
         },
       ]
     }
-    terraform_addons = {
-      providers = [
-        {
-          name = "google",
-          version_constraints = ">=3.0, <= 3.71"
-        },
-        {
-          name = "google-beta",
-          version_constraints = "~>3.50"
-        },
-        {
-          name = "null",
-          version_constraints = "~> 3.0"
-        }
-      ]
-      raw_config = <<EOF
-resource "random_password" "db" {
-  length = 16
-  special = true
-}
-EOF
+    deployment = {
```

Comment:
This is breaking change for all users. I don't think we should require everyone to change their config in order to fix this bug. The fix should be silently in our logic without user-facing changes.

Reply:
In that case, it seems we can hardcode this in dumpTemplate. i.e. when ti.Name = `project` do sth...

any thoughts?

Reply:
Did you figure out why resources recipe works but not deployment recipe? We didn't need to hardcode `ti.Name=project ...` in order to make that work.

Reply:
the reason is essentially this:

1. we need to pass the flattened data by key `terraform_addons` (almost) so that `ti.Data' wouldn't be empty for the validation (when I showed you guys yesterday I turned on the printing statement at another place which was not what I meant to...)
2. but passing by `terraform_addons` has a problem that it passes all the value `terraform_addons` holds as a key in `depolyment`, namely we lose the `terraform_addons` itself as the key of the map in the validation layer so that the validation process doesn't recognize it. If you compare to `resources`, we pass by the key `resources` so that all the actual key of the `resources` get passed down to validation as opposed to (`providers`, `raw_config` etc subfields of `terraform_addons`) -- sorry a little bit clunky hopefully it makes sense
3. the most straightforward way to salvage this is wrapping all the `terraform_addons` in `deployment` and pass by key `deployment` so on the next level it would have the desired key `terraform_addons` to validate against. 
4. But since you pointed out we would't want to entail such breaking change we may have to hardcode it in `dumpTemplate`. Something I can think of is when `ti.Name == "project"`, create another data carrier `d` and  make `d["terraform_addons"]` carry all the scattered fields in `ti.Data` that used to belong to `terraform_addons`. I know this is grotesque though

Reply:
I'm not sure I understand what the concern is. It seems to me that if we add a new function (passthrough) and only use it here in this one place (project.hcl) in order to pass terraform_addons to deployment.hcl, then it won't affect any other recipes.

Reply:
I played with it a little bit. See https://github.com/GoogleCloudPlatform/healthcare-data-protection-suite/pull/991.

Reply:
> I'm not sure I understand what the concern is. It seems to me that if we add a new function (passthrough) and only use it here in this one place (project.hcl) in order to pass terraform_addons to deployment.hcl, then it won't affect any other recipes.

I got wrong what you meant. yeah that should work too

Reply:
passthrough sgtm as well.